Sources/ApolloCodegenLib/IR/IR+RootFieldBuilder.swift

```diff
@@ -241,7 +262,11 @@ extension IR {
       let type = parentTypePath.parentType == conditionalSelectionSet.parentType ?
       nil : conditionalSelectionSet.parentType
 
-      return ScopeCondition(type: type, conditions: inclusionResult.conditions)
+      return ScopeCondition(
+        type: type,
+        conditions: inclusionResult.conditions,
+        isDeferred: IsDeferred(compilationResult: isDeferred)
+      )
```

Comment:
These changes also should really be tested in tests on the `RootFieldBuilder`, not just the `SelectionSetTemplateTests`.